Adjust timeline

LDSamson opened the pull request:
PR is part of the process of adding more items in the timeline, such as concomitant medication and/or concomitant procedures. 
Will help to solve this: #12 and some of this: #221

Two main changes here: 

- Places the timeline in the header instead of in mod_common forms
- Adds a toggle to optionally show the timeline

Timeline is now on by default when viewing common forms and off when viewing study forms.

https://github.com/user-attachments/assets/6947dc67-a46c-47a5-ad02-c51ab95360b1

LDSamson committed:
detach timeline from common_forms

LDSamson committed:
Merge branch 'dev' into ls_adjust_timeline

LDSamson committed:
Add cursor hovers for header widgets to indicate that they are clickable

LDSamson committed:
Revert adding sidebar in mod_timeline

LDSamson committed:
Start adding buttons in mod_study_forms and mod_common_forms for controlling stop widgets

LDSamson committed:
Update logic to only change timeline and not headers

LDSamson committed:
update input IDs

LDSamson committed:
Ùse tagList again around timeline

LDSamson committed:
Undo aligning changes

LDSamson committed:
Move added code together

LDSamson committed:
Hide toggle in form level review

LDSamson committed:
Move timeline to header

LDSamson committed:
Move logic to module

LDSamson committed:
remove timeline data from mod_common_forms and update docs

LDSamson committed:
Ensure toggles reactive toggles are always there with default values to avoid errors

LDSamson committed:
Update test ID for timeline

LDSamson committed:
Update tests and snaps

LDSamson committed:
Update snaps

LDSamson committed:
Update tests

LDSamson committed:
Bit more robust timeline data

LDSamson committed:
Remove scenario in mod_common_forms involving timeline

LDSamson committed:
Put navinfo objects together

LDSamson committed:
Merge branch 'dev' into ls_adjust_timeline

LDSamson committed:
Bump version

LDSamson committed:
Simplify feature 3 snaps

LDSamson committed:
Fix docs

LDSamson committed:
Add test for timeline toggle

LDSamson committed:
Merge branch 'dev' into ls_adjust_timeline

# Conflicts:
#	DESCRIPTION
#	R/app_server.R
#	R/mod_common_forms.R
#	R/mod_header_widgets.R
#	inst/golem-config.yml
#	man/mod_common_forms_server.Rd
#	man/mod_header_widgets_server.Rd
#	tests/testthat/test-app_feature_03.R
#	tests/testthat/test-mod_common_forms.R
#	tests/testthat/test-mod_header_widgets.R

LDSamson committed:
Update CSS for new timeline toggle

jthompson-arcus reviewed (commented):
Mostly focusing on visuals for this review. Didn't have time to dig into the code.

- At first I thought the toggle should be above the timeline but I think it makes the most sense to have it as you do as part of the nav bar. I do feel like the toggle is slightly too big and feels awkward.
- I find turning the timeline off and on a bit jarring. I think some transitions need to be added but that's a bit more involved.

@aclark02-arcus what do you think?

aclark02-arcus commented:
Agree with Jeff's comments & adding one more. I don't think the toggle is too big... but I don't think it's obvious what it does just by looking at it:

<img width="56" height="26" alt="image" src="https://github.com/user-attachments/assets/535abce6-7e66-4839-8c4f-feb8c323b306" />

It's really hard to tell what that icon is displaying. Would it be too crazy to just include a "Timeline" label next to the toggle?

LDSamson committed:
Change switch

LDSamson committed:
Add animation

LDSamson committed:
code styling

LDSamson committed:
Merge branch 'dev' into ls_adjust_timeline

LDSamson committed:
Bump version

LDSamson committed:
Add news

LDSamson committed:
Add wait time in test to wait for animation

LDSamson commented:
Thanks for the suggestions. I added an animation and changed the toggle, see changes below. 
@aclark02-arcus and @jthompson-arcus what do you think?
https://github.com/user-attachments/assets/2892a88f-e466-4fbd-84d8-7bbbf35209bc

aclark02-arcus commented:
@LDSamson, that looks real nice. I only hope people don't confuse it with the static visit timeline above.

jthompson-arcus reviewed (commented):
This looks good to me. Just need to sync up with the `dev` branch.

LDSamson committed:
Merge branch 'dev' into ls_adjust_timeline

LDSamson committed:
Bump version

jthompson-arcus reviewed (approved):
LGTM
